docs/help/contributor/style-guide/documenting-code.md

```diff
@@ -0,0 +1,212 @@
+# Documenting Code
+
+??? tip "Looking for a template?"
+    //todo
+
+## Documenting Commands
+
+>**The standard line is “X the Y by running:”**
+
+>It meets these criteria:
+
+>* Explicitly mentions running the command (this isn’t always obvious)
+* **Uses “run”** (and not “type”, “execute”, etc -- we want consistency)
+* Starts with the key information that describes the command, e.g. “To do X...enter Y command:”, “Do X by entering Y command:”
+* As short as possible
+
+> If you must deviate from the standard line, ensure you still meet the above criteria.
+
+=== ":white_check_mark: Correct"
+    Create the service by running:
+    ```bash
+    kn create service <service-name>
+    ```
+    Where `<service-name>` is the name of your Knative Service.
+
+=== ":no_entry: Incorrect"
+    **Example 1:**
+
+    Create the service:
+    ```bash
+    kn create service <service-name>
+    ```
+    Where `<service-name>` is the name of your Knative Service.
+
+    **Example 2:**
+
+    Run:
+    ```bash
+    kn create service <service-name>
+    ```
+    Where `<service-name>` is the name of your Knative Service.
+
+    **Example 3:**
+
+    Run the command below to create a service:
+
+    ```bash
+    kn create service <service-name>
+    ```
+    Where `<service-name>` is the name of your Knative Service.
+
+
+## Documenting YAML
+>Use **kubectl apply** for files/objects that the user creates — it works for both “create” and “update”, and the source of truth is their local files.
+Use **kubectl edit** for files which are shipped as part of the Knative software, like the serving/eventing ConfigMaps.
+
+=== ":white_check_mark: Correct"
+
+    Creating a new file:
+    ```
+    kubectl apply -f - <<EOF
+    # code
+    EOF
+    ```
+
+    Editing a file:
+    ```
+    kubectl -n <namespace> edit cm <filename>
+    ```
```

Comment:
I think this is specific to configmaps but I'm not sure; @vaikas ?
I think `apply` can be used to create or edit regular YAML files?
Might be wrong just thought I'd check.

Reply:
This is related when editing config maps that already exist in the system.
 
It should not use `file` instead you are editing a configmap resource

Editing a configmap
```
kubectl -n <namespace> edit configmap RESOURCE/NAME
```

This is closer on how `kubectl` help docs

Reply:
+1 @omerbensaadon I think we need to call out the difference here between the kubectl command for editing/creating other YAMLs and resources vs this specific one for modifying existing configmaps.

Reply:
Added suggestions above

Reply:
thank you @abrennan89 !